src/vm/exceptionhandling.cpp

```diff
@@ -3649,12 +3649,57 @@ ExceptionTracker* ExceptionTracker::GetOrCreateTracker(
             {
                 if (fIsInterleavedHandling)
                 {
-                    // Remember part of the current state that needs to be transferred to
-                    // the newly created tracker.
-                    previousTrackerPartialState.Save(pTracker);
-
-                    // We just transitioned from 2nd pass to 1st pass when we handle the exception in an interleaved manner
+                    // We just transitioned from 2nd pass to 1st pass when we handle the exception in an interleaved manner.
+                    // In interleaved exception handling a single exception is handled in an interleaved
+                    // series of first and second passes. We re-create the exception tracker in-place
+                    // and carry over several members that were set when processing one of the previous frames.
                     EH_LOG((LL_INFO100, ">>continued processing of PREVIOUS exception (interleaved handling)\n"));
+                    {
+                        GCX_COOP();
+
+                        // Remember part of the current state that needs to be transferred to
+                        // the newly created tracker.
+                        previousTrackerPartialState.Save(pTracker);
+
+                        // Rember the previous tracker pointer so that we can restore it
+                        ExceptionTracker* prevTracker = pTracker->m_pPrevNestedInfo;
+
+                        // Free the tracker resources so that we can recreate a new one in-place
+                        pTracker->ReleaseResources();
+
+                        new (pTracker) ExceptionTracker(ControlPc,
+                                                        pExceptionRecord,
+                                                        pContextRecord);
+
+                        // Restore part of the tracker state that was saved before recreating the tracker
+                        previousTrackerPartialState.Restore(pTracker);
```

Comment:
Delete call to previousTrackerPartialState.Restore towards the end of this method? It seems to be unused now that the code was moved to here. It may be also nice to move previousTrackerPartialState definition to this scope.